test/unittest/blas1/blas1_iamax_test.cpp

```diff
@@ -31,24 +32,20 @@ void run_test(const combination_t<scalar_t> combi) {
   blas::SB_Handle sb_handle(q);
 
   // Iterators
-  auto gpu_x_v = blas::make_sycl_iterator_buffer<scalar_t>(int(size * incX));
-  blas::helper::copy_to_device(sb_handle.get_queue(), x_v.data(), gpu_x_v,
-                               size * incX);
-  auto gpu_out_s = blas::make_sycl_iterator_buffer<tuple_t>(int(1));
-  blas::helper::copy_to_device(sb_handle.get_queue(), &out_s, gpu_out_s, 1);
-
-  _iamax(sb_handle, size, gpu_x_v, incX, gpu_out_s);
-  auto event =
-      blas::helper::copy_to_host(sb_handle.get_queue(), gpu_out_s, &out_s, 1);
-  sb_handle.wait(event);
-
-  using data_tuple_t = IndexValueTuple<int, scalar_t>;
-  data_tuple_t out_data_s{out_s.ind, static_cast<scalar_t>(out_s.val)};
+  auto gpu_x_v = blas::make_sycl_iterator_buffer<scalar_t>(x_v, size * incX);
+
+  if (api == api_type::async) {
+    auto gpu_out_s = blas::make_sycl_iterator_buffer<tuple_t>(&out_s, 1);
```

Comment:
I'm not entirely sure why this function returns a tuple on the async version and a float on the sync version. Seems inconsistent. Regardless, we should probably still assert on out_s.data for the async version.

Same comment would apply for iamin